# Proceso interno de nuevos cambios 

In [1]:
import pandas as pd
import numpy as np

In [ ]:
[3297269,
 6504609,
 7530382,
 15895374,
 17412049,
 17591769,
 18435275,
 19252859,
 20438765,
 23738879,
 28814071,
 28837359,
 38233613,
 40315034,
 41783045,
 65496684,
 79245518,
 79314863]

In [2]:
df = pd.read_excel('datos_CAIMEDv7.xlsx')
df

,individuo,fecha_constetimiento_informado,fecha_de_nacimiento,edad_calculada,estado_civil,sexo,lugar_de_nacimiento,lugar_de_procedencia,escolaridad,estrato_socieconomico,...,Antidiabético,¿se_hizo_examen?,¿resultados_relacionados_a_obesidad?,¿son_signficativos_relacionado_a_obesidad?,¿resultados_relacionados_a_problemas cardiacos?,¿son_signficativos_relacionado_a_problemas cardiacos?,¿resultados_relacionados_a_tension arterial?,¿son_signficativos_relacionado_a_tension arterial?,¿resultados_relacionados_a_accidente cerebrovascular?,¿son_signficativos_relacionado_a_accidente cerebrovascular?
0,70033505,2023-08-31,1951-09-10 00:00:00,71.0,casado,masculino,bogota,bogota,profesional,6.0,...,no,si,si,si,no,no,no,no,no,no
1,41760470,2023-08-30,1955-11-22 00:00:00,67.0,divorciado,femenino,bogota,bogota,profesional,3.0,...,no,si,no,no,no,no,no,no,no,no
2,17171452,2023-08-30,1945-11-09 00:00:00,77.0,casado,masculino,jesus,bogota,especialista,4.0,...,no,si,no,no,no,no,no,no,no,no
3,1019032057,2023-08-25,1989-12-03 00:00:00,33.0,union libre,femenino,barranquilla,bogota,tecnologo,4.0,...,no,si,si,si,no,no,si,si,no,no
4,19250084,2023-08-24,1954-12-01 00:00:00,68.0,casado,masculino,medellin,bogota,profesional,6.0,...,no,si,si,si,si,no,no,no,no,no
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1002,7561381,2023-05-17,1970-07-09 00:00:00,52.0,soltero,masculino,armenia,armenia,secundaria completa,2.0,...,no,no,no,no,no,no,no,no,no,no
1003,41899611,2023-05-17,1954-11-10 00:00:00,68.0,union libre,femenino,armenia,armenia,primaria incompleta,2.0,...,no,no,no,no,no,no,no,no,no,no
1004,19317507,2023-05-17,1956-11-05 00:00:00,66.0,casado,masculino,bogota,circasia,especialista,3.0,...,no,no,no,no,no,no,no,no,no,no
1005,4398004,2023-05-16,1942-01-24 00:00:00,81.0,casado,masculino,montenegro,armenia,primaria completa,4.0,...,si,no,no,no,no,no,no,no,no,no


- **IMC**: Se utilizará la columna de **IMC calculada** (resaltada en verde, basada en valores de peso y talla en la base de datos).
    - Se excluyen del análisis los registros con diferencias en IMC mayores de 2 o menores de -2:
      - ID: `20438765`, `17591769`, `41783045`, `28837359`, `3297269`, `79314863`.

In [3]:
sub_temp = []
index_drop = []
for Id in [20438765, 17591769, 41783045, 28837359, 3297269, 79314863]:
    rows = df[df['individuo'] == Id]
    indices = rows.index.tolist()
    index_drop.append(indices)
    sub_temp.append(rows)
    df= df.drop(int(indices[0]))
    print(f"Indices para Id {Id}: {indices}")
result = pd.concat(sub_temp, ignore_index=False)
result


Indices para Id 20438765: [121]
Indices para Id 17591769: [161]
Indices para Id 41783045: [9]
Indices para Id 28837359: [518]
Indices para Id 3297269: [158]
Indices para Id 79314863: [132]


,individuo,fecha_constetimiento_informado,fecha_de_nacimiento,edad_calculada,estado_civil,sexo,lugar_de_nacimiento,lugar_de_procedencia,escolaridad,estrato_socieconomico,...,Antidiabético,¿se_hizo_examen?,¿resultados_relacionados_a_obesidad?,¿son_signficativos_relacionado_a_obesidad?,¿resultados_relacionados_a_problemas cardiacos?,¿son_signficativos_relacionado_a_problemas cardiacos?,¿resultados_relacionados_a_tension arterial?,¿son_signficativos_relacionado_a_tension arterial?,¿resultados_relacionados_a_accidente cerebrovascular?,¿son_signficativos_relacionado_a_accidente cerebrovascular?
121,20438765,2023-05-17,1950-01-08 00:00:00,73.0,soltero,femenino,caqueza,bogota,secundaria incompleta,3.0,...,no,no,no,no,no,no,no,no,no,no
161,17591769,2023-08-10,1977-02-04 00:00:00,46.0,union libre,masculino,puerto,paz,secundaria incompleta,2.0,...,no,no,no,no,no,no,no,no,no,no
9,41783045,2023-08-18,1959-10-27 00:00:00,63.0,soltero,femenino,bogota,bogota,tecnologo,3.0,...,no,si,si,si,no,no,no,no,no,no
518,28837359,2023-07-07,1957-06-22 00:00:00,66.0,viudo,femenino,mariquita,mariquita,primaria incompleta,1.0,...,no,no,no,no,no,no,no,no,no,no
158,3297269,2023-08-17,1949-03-25 00:00:00,74.0,viudo,masculino,socota,yopal,primaria incompleta,1.0,...,no,no,no,no,no,no,no,no,no,no
132,79314863,2023-05-17,1963-09-22 00:00:00,59.0,union libre,femenino,bogota,bogota,secundaria completa,2.0,...,no,si,no,no,no,no,no,no,no,no


In [4]:
df.reset_index(drop=True)

,individuo,fecha_constetimiento_informado,fecha_de_nacimiento,edad_calculada,estado_civil,sexo,lugar_de_nacimiento,lugar_de_procedencia,escolaridad,estrato_socieconomico,...,Antidiabético,¿se_hizo_examen?,¿resultados_relacionados_a_obesidad?,¿son_signficativos_relacionado_a_obesidad?,¿resultados_relacionados_a_problemas cardiacos?,¿son_signficativos_relacionado_a_problemas cardiacos?,¿resultados_relacionados_a_tension arterial?,¿son_signficativos_relacionado_a_tension arterial?,¿resultados_relacionados_a_accidente cerebrovascular?,¿son_signficativos_relacionado_a_accidente cerebrovascular?
0,70033505,2023-08-31,1951-09-10 00:00:00,71.0,casado,masculino,bogota,bogota,profesional,6.0,...,no,si,si,si,no,no,no,no,no,no
1,41760470,2023-08-30,1955-11-22 00:00:00,67.0,divorciado,femenino,bogota,bogota,profesional,3.0,...,no,si,no,no,no,no,no,no,no,no
2,17171452,2023-08-30,1945-11-09 00:00:00,77.0,casado,masculino,jesus,bogota,especialista,4.0,...,no,si,no,no,no,no,no,no,no,no
3,1019032057,2023-08-25,1989-12-03 00:00:00,33.0,union libre,femenino,barranquilla,bogota,tecnologo,4.0,...,no,si,si,si,no,no,si,si,no,no
4,19250084,2023-08-24,1954-12-01 00:00:00,68.0,casado,masculino,medellin,bogota,profesional,6.0,...,no,si,si,si,si,no,no,no,no,no
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
996,7561381,2023-05-17,1970-07-09 00:00:00,52.0,soltero,masculino,armenia,armenia,secundaria completa,2.0,...,no,no,no,no,no,no,no,no,no,no
997,41899611,2023-05-17,1954-11-10 00:00:00,68.0,union libre,femenino,armenia,armenia,primaria incompleta,2.0,...,no,no,no,no,no,no,no,no,no,no
998,19317507,2023-05-17,1956-11-05 00:00:00,66.0,casado,masculino,bogota,circasia,especialista,3.0,...,no,no,no,no,no,no,no,no,no,no
999,4398004,2023-05-16,1942-01-24 00:00:00,81.0,casado,masculino,montenegro,armenia,primaria completa,4.0,...,si,no,no,no,no,no,no,no,no,no


- **Edad**: Se utilizará la columna de **edad calculada** (resaltada en verde).
    - Para los tres registros con diferencias en la edad calculada, la **fecha de nacimiento fue verificada** en la base de datos de vacunación (PAIWEB).

- **Presión Arterial Sistólica y Diastólica**: 
    - Se excluyen del análisis los registros en los que la diferencia entre presión arterial sistólica y diastólica es inferior a 20 o superior a 90.
      - ID: `23738879`, `19252859`, `18435275`, `15895374`, `28814071`, `6504609`, `38233613`, `4000000` (resaltados en rojo).

In [5]:
sub_temp = []
index_drop = []
for Id in [23738879, 19252859, 18435275, 15895374, 28814071, 6504609, 38233613, 4000000]:
    rows = df[df['individuo'] == Id]
    indices = rows.index.tolist()
    index_drop.append(indices)
    sub_temp.append(rows)
    df= df.drop(int(indices[0]))
    print(f"Indices para Id {Id}: {indices}")
result = pd.concat(sub_temp, ignore_index=False)
result

Indices para Id 23738879: [244]
Indices para Id 19252859: [282]
Indices para Id 18435275: [927]
Indices para Id 15895374: [948]
Indices para Id 28814071: [534]
Indices para Id 6504609: [314]
Indices para Id 38233613: [524]
Indices para Id 4000000: [635]


,individuo,fecha_constetimiento_informado,fecha_de_nacimiento,edad_calculada,estado_civil,sexo,lugar_de_nacimiento,lugar_de_procedencia,escolaridad,estrato_socieconomico,...,Antidiabético,¿se_hizo_examen?,¿resultados_relacionados_a_obesidad?,¿son_signficativos_relacionado_a_obesidad?,¿resultados_relacionados_a_problemas cardiacos?,¿son_signficativos_relacionado_a_problemas cardiacos?,¿resultados_relacionados_a_tension arterial?,¿son_signficativos_relacionado_a_tension arterial?,¿resultados_relacionados_a_accidente cerebrovascular?,¿son_signficativos_relacionado_a_accidente cerebrovascular?
244,23738879,2023-06-10,1943-11-05 00:00:00,79.0,viudo,femenino,tamara,yopal,secundaria incompleta,2.0,...,no,no,no,no,no,no,no,no,no,no
282,19252859,2023-05-24,1954-10-23 00:00:00,68.0,union libre,masculino,villavicencio,yopal,secundaria incompleta,1.0,...,si,si,no,no,no,no,no,no,no,no
927,18435275,2023-06-06,1972-01-01 00:00:00,51.0,casado,masculino,armenia,armenia,secundaria completa,2.0,...,no,si,si,no,no,no,no,no,no,no
948,15895374,2023-06-03,1950-01-16 00:00:00,73.0,casado,masculino,libano,armenia,secundaria incompleta,3.0,...,no,no,no,no,no,no,no,no,no,no
534,28814071,2023-07-04,1954-03-31 00:00:00,69.0,divorciado,femenino,libano,ibague,primaria incompleta,3.0,...,si,no,no,no,no,no,no,no,no,no
314,6504609,2023-08-15,1946-01-17 00:00:00,77.0,casado,masculino,el tambo,acacias,primaria incompleta,2.0,...,no,no,no,no,no,no,no,no,no,no
524,38233613,2023-07-10,1953-10-06 00:00:00,69.0,casado,femenino,madrid,ibague,primaria completa,2.0,...,no,si,no,no,no,no,no,no,no,no
635,4000000,2023-06-06,NaT,NaN,NaN,NaN,NaN,ibague,NaN,NaN,...,no,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [6]:
df.reset_index(drop=True)

,individuo,fecha_constetimiento_informado,fecha_de_nacimiento,edad_calculada,estado_civil,sexo,lugar_de_nacimiento,lugar_de_procedencia,escolaridad,estrato_socieconomico,...,Antidiabético,¿se_hizo_examen?,¿resultados_relacionados_a_obesidad?,¿son_signficativos_relacionado_a_obesidad?,¿resultados_relacionados_a_problemas cardiacos?,¿son_signficativos_relacionado_a_problemas cardiacos?,¿resultados_relacionados_a_tension arterial?,¿son_signficativos_relacionado_a_tension arterial?,¿resultados_relacionados_a_accidente cerebrovascular?,¿son_signficativos_relacionado_a_accidente cerebrovascular?
0,70033505,2023-08-31,1951-09-10 00:00:00,71.0,casado,masculino,bogota,bogota,profesional,6.0,...,no,si,si,si,no,no,no,no,no,no
1,41760470,2023-08-30,1955-11-22 00:00:00,67.0,divorciado,femenino,bogota,bogota,profesional,3.0,...,no,si,no,no,no,no,no,no,no,no
2,17171452,2023-08-30,1945-11-09 00:00:00,77.0,casado,masculino,jesus,bogota,especialista,4.0,...,no,si,no,no,no,no,no,no,no,no
3,1019032057,2023-08-25,1989-12-03 00:00:00,33.0,union libre,femenino,barranquilla,bogota,tecnologo,4.0,...,no,si,si,si,no,no,si,si,no,no
4,19250084,2023-08-24,1954-12-01 00:00:00,68.0,casado,masculino,medellin,bogota,profesional,6.0,...,no,si,si,si,si,no,no,no,no,no
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
988,7561381,2023-05-17,1970-07-09 00:00:00,52.0,soltero,masculino,armenia,armenia,secundaria completa,2.0,...,no,no,no,no,no,no,no,no,no,no
989,41899611,2023-05-17,1954-11-10 00:00:00,68.0,union libre,femenino,armenia,armenia,primaria incompleta,2.0,...,no,no,no,no,no,no,no,no,no,no
990,19317507,2023-05-17,1956-11-05 00:00:00,66.0,casado,masculino,bogota,circasia,especialista,3.0,...,no,no,no,no,no,no,no,no,no,no
991,4398004,2023-05-16,1942-01-24 00:00:00,81.0,casado,masculino,montenegro,armenia,primaria completa,4.0,...,si,no,no,no,no,no,no,no,no,no


- **cLDL (mg/dL)**: 
    - Tres registros con valores de `2` y `6` fueron excluidos del análisis.
      - ID: `19062972`, `65496684`, `7530382` (resaltados en rojo).

In [7]:
sub_temp = []
index_drop = []
for Id in [19062972, 65496684, 7530382]:
    rows = df[df['individuo'] == Id]
    indices = rows.index.tolist()
    index_drop.append(indices)
    sub_temp.append(rows)
    df= df.drop(int(indices[0]))
    print(f"Indices para Id {Id}: {indices}")
result = pd.concat(sub_temp, ignore_index=False)
result

Indices para Id 19062972: [858]
Indices para Id 65496684: [19]
Indices para Id 7530382: [705]


,individuo,fecha_constetimiento_informado,fecha_de_nacimiento,edad_calculada,estado_civil,sexo,lugar_de_nacimiento,lugar_de_procedencia,escolaridad,estrato_socieconomico,...,Antidiabético,¿se_hizo_examen?,¿resultados_relacionados_a_obesidad?,¿son_signficativos_relacionado_a_obesidad?,¿resultados_relacionados_a_problemas cardiacos?,¿son_signficativos_relacionado_a_problemas cardiacos?,¿resultados_relacionados_a_tension arterial?,¿son_signficativos_relacionado_a_tension arterial?,¿resultados_relacionados_a_accidente cerebrovascular?,¿son_signficativos_relacionado_a_accidente cerebrovascular?
858,19062972,2023-07-06,1946-05-23 00:00:00,77.0,soltero,masculino,quimbaya,quimbaya,primaria incompleta,3.0,...,si,si,si,no,no,no,no,no,no,no
19,65496684,2023-08-02,1960-03-11 00:00:00,63.0,soltero,femenino,armero,bogota,profesional,4.0,...,no,si,si,si,no,no,no,no,no,no
705,7530382,2023-12-04,1959-10-21 00:00:00,64.0,union libre,masculino,pereira,armenia,primaria completa,2.0,...,no,si,si,si,no,no,si,si,no,no


In [8]:
df.reset_index(drop=True)

,individuo,fecha_constetimiento_informado,fecha_de_nacimiento,edad_calculada,estado_civil,sexo,lugar_de_nacimiento,lugar_de_procedencia,escolaridad,estrato_socieconomico,...,Antidiabético,¿se_hizo_examen?,¿resultados_relacionados_a_obesidad?,¿son_signficativos_relacionado_a_obesidad?,¿resultados_relacionados_a_problemas cardiacos?,¿son_signficativos_relacionado_a_problemas cardiacos?,¿resultados_relacionados_a_tension arterial?,¿son_signficativos_relacionado_a_tension arterial?,¿resultados_relacionados_a_accidente cerebrovascular?,¿son_signficativos_relacionado_a_accidente cerebrovascular?
0,70033505,2023-08-31,1951-09-10 00:00:00,71.0,casado,masculino,bogota,bogota,profesional,6.0,...,no,si,si,si,no,no,no,no,no,no
1,41760470,2023-08-30,1955-11-22 00:00:00,67.0,divorciado,femenino,bogota,bogota,profesional,3.0,...,no,si,no,no,no,no,no,no,no,no
2,17171452,2023-08-30,1945-11-09 00:00:00,77.0,casado,masculino,jesus,bogota,especialista,4.0,...,no,si,no,no,no,no,no,no,no,no
3,1019032057,2023-08-25,1989-12-03 00:00:00,33.0,union libre,femenino,barranquilla,bogota,tecnologo,4.0,...,no,si,si,si,no,no,si,si,no,no
4,19250084,2023-08-24,1954-12-01 00:00:00,68.0,casado,masculino,medellin,bogota,profesional,6.0,...,no,si,si,si,si,no,no,no,no,no
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
985,7561381,2023-05-17,1970-07-09 00:00:00,52.0,soltero,masculino,armenia,armenia,secundaria completa,2.0,...,no,no,no,no,no,no,no,no,no,no
986,41899611,2023-05-17,1954-11-10 00:00:00,68.0,union libre,femenino,armenia,armenia,primaria incompleta,2.0,...,no,no,no,no,no,no,no,no,no,no
987,19317507,2023-05-17,1956-11-05 00:00:00,66.0,casado,masculino,bogota,circasia,especialista,3.0,...,no,no,no,no,no,no,no,no,no,no
988,4398004,2023-05-16,1942-01-24 00:00:00,81.0,casado,masculino,montenegro,armenia,primaria completa,4.0,...,si,no,no,no,no,no,no,no,no,no


- **Colesterol**: 
    - Se realizó una validación del reporte de colesterol calculando la diferencia entre el colesterol total y la suma de colesterol HDL + LDL.
    - Se encontraron 9 registros sin datos de colesterol y 3 registros en los que el valor era igual o menor a cero, por lo que fueron excluidos del análisis.
      - ID: `79245518`, `17412049`, `40315034` (resaltados en rojo).

In [9]:
sub_temp = []
index_drop = []
for Id in [79245518, 17412049, 40315034]:
    rows = df[df['individuo'] == Id]
    indices = rows.index.tolist()
    index_drop.append(indices)
    sub_temp.append(rows)
    df= df.drop(int(indices[0]))
    print(f"Indices para Id {Id}: {indices}")
result = pd.concat(sub_temp, ignore_index=False)
result

Indices para Id 79245518: [49]
Indices para Id 17412049: [302]
Indices para Id 40315034: [323]


,individuo,fecha_constetimiento_informado,fecha_de_nacimiento,edad_calculada,estado_civil,sexo,lugar_de_nacimiento,lugar_de_procedencia,escolaridad,estrato_socieconomico,...,Antidiabético,¿se_hizo_examen?,¿resultados_relacionados_a_obesidad?,¿son_signficativos_relacionado_a_obesidad?,¿resultados_relacionados_a_problemas cardiacos?,¿son_signficativos_relacionado_a_problemas cardiacos?,¿resultados_relacionados_a_tension arterial?,¿son_signficativos_relacionado_a_tension arterial?,¿resultados_relacionados_a_accidente cerebrovascular?,¿son_signficativos_relacionado_a_accidente cerebrovascular?
49,79245518,2023-06-23,1969-07-15 00:00:00,53.0,casado,masculino,bogota,bogota,profesional,5.0,...,no,si,si,si,no,no,no,no,no,no
302,17412049,2023-09-01,1963-06-18 00:00:00,60.0,soltero,masculino,acacias,acacias,pre-escolar incompleta,1.0,...,si,no,no,no,no,no,no,no,no,no
323,40315034,2023-07-28,1957-03-16 00:00:00,66.0,viudo,femenino,yarumal,acacias,primaria incompleta,2.0,...,si,no,no,no,no,no,no,no,no,no


## Datos luego de extracciones:
    Finalmete se mantienen 987 regsitros 

In [10]:
df = df.reset_index(drop=True)
df

,individuo,fecha_constetimiento_informado,fecha_de_nacimiento,edad_calculada,estado_civil,sexo,lugar_de_nacimiento,lugar_de_procedencia,escolaridad,estrato_socieconomico,...,Antidiabético,¿se_hizo_examen?,¿resultados_relacionados_a_obesidad?,¿son_signficativos_relacionado_a_obesidad?,¿resultados_relacionados_a_problemas cardiacos?,¿son_signficativos_relacionado_a_problemas cardiacos?,¿resultados_relacionados_a_tension arterial?,¿son_signficativos_relacionado_a_tension arterial?,¿resultados_relacionados_a_accidente cerebrovascular?,¿son_signficativos_relacionado_a_accidente cerebrovascular?
0,70033505,2023-08-31,1951-09-10 00:00:00,71.0,casado,masculino,bogota,bogota,profesional,6.0,...,no,si,si,si,no,no,no,no,no,no
1,41760470,2023-08-30,1955-11-22 00:00:00,67.0,divorciado,femenino,bogota,bogota,profesional,3.0,...,no,si,no,no,no,no,no,no,no,no
2,17171452,2023-08-30,1945-11-09 00:00:00,77.0,casado,masculino,jesus,bogota,especialista,4.0,...,no,si,no,no,no,no,no,no,no,no
3,1019032057,2023-08-25,1989-12-03 00:00:00,33.0,union libre,femenino,barranquilla,bogota,tecnologo,4.0,...,no,si,si,si,no,no,si,si,no,no
4,19250084,2023-08-24,1954-12-01 00:00:00,68.0,casado,masculino,medellin,bogota,profesional,6.0,...,no,si,si,si,si,no,no,no,no,no
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
982,7561381,2023-05-17,1970-07-09 00:00:00,52.0,soltero,masculino,armenia,armenia,secundaria completa,2.0,...,no,no,no,no,no,no,no,no,no,no
983,41899611,2023-05-17,1954-11-10 00:00:00,68.0,union libre,femenino,armenia,armenia,primaria incompleta,2.0,...,no,no,no,no,no,no,no,no,no,no
984,19317507,2023-05-17,1956-11-05 00:00:00,66.0,casado,masculino,bogota,circasia,especialista,3.0,...,no,no,no,no,no,no,no,no,no,no
985,4398004,2023-05-16,1942-01-24 00:00:00,81.0,casado,masculino,montenegro,armenia,primaria completa,4.0,...,si,no,no,no,no,no,no,no,no,no
